kotlin/lib/src/main/kotlin/goodmetrics/Metrics.kt

```diff
@@ -8,20 +8,27 @@ class Metrics internal constructor(
     internal var timestampMillis: Long,
     internal val startNanoTime: Long,
 ) {
+    sealed interface Dimension {
+        val name: String
+        val value: Any
+        data class StringDimension(override val name: String, override val value: String): Dimension
+        data class NumberDimension(override val name: String, override val value: Long): Dimension
+        data class BooleanDimension(override val name: String, override val value: Boolean): Dimension

```

Comment:
nit: these are all already namespaced under `Dimension`. It's pointlessly redundant to repeat Dimension before and after the name. `Dimension.String` is less line noise than `Dimension.StringDimension`

Reply:
Right, so if we try to simply name it as `String` the Kotlin compiler will presume all future parameters labeled `String` will be a `Metrics.Dimension.String`, requiring us to explicitly name all `String` parameters in the file as a `kotlin.String`. 

I guess I could always name them to be shorthand, like `Metrics.Dimension.Str` or `Metrics.Dimension.Bool`